### PR TITLE
new sub-TLV: LPP, flags

### PR DESCRIPTION
**Pull request by Bruno007**
Addition of 2 sub-TLV:
- LPP: number of LSP received before sending a PSNP
- Flags
- Flag: ordered ack

### CODE REVIEW

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 1]
```diff
@@ -346,113 +356,138 @@
...
-				<t>Length: 4 octets</t>
-				<t>Value: minimum interval, in micro-seconds, between two consecutive LSPs sent after the burst window has been used</t>
-				<t>The LSP Transmission Interval is an advertisement of the receiver's steady-state LSP reception rate.</t>
+			<section anchor="Flags" title="Flags sub-TLV">
+				<t>The sub-TLV Flags advertises a set of flags. Flags are standardized and IANA registered.</t>
+				<t>Type: 4</t>
+				<t>Length: Indicates the length in octets (1-8) of the Value field. The length SHOULD be the minimum required to send all bits that are set.</t>
+				<t>Value: List of flags.</t>
+
+				<artwork align="left" pn="Flags">
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:386]: This needs to be wrapped in t and figure tags.  Xml2rfc does not understand 'pn'.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 2]
```diff
@@ -202,136 +203,145 @@
...
 
-      <t>
+			<t>
 	Improving the LSP flooding rate is complementary to protocol
 	extensions that reduce LSP flooding traffic by reducing the
 	flooding topology such as Mesh Groups <xref target="RFC2973"/>
-	or Dynamic Flooding <xref
-	target="I-D.ietf-lsr-dynamic-flooding"/>. Reduction of the
+	or Dynamic Flooding <xref target="I-D.ietf-lsr-dynamic-flooding"/>
+. Reduction of the
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:214]: This re-filling of the paragraph introduces an incorrect space after the reference.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 3]
```diff
@@ -346,113 +356,138 @@
...
+		</section>
+		<section anchor="InterfaceLSPTransmissionInterval" title="LSP Transmission Interval sub-TLV">
+			<t>The LSP Transmission Interval sub-TLV advertises the minimum interval, in micro-seconds, between LSPs arrivals which can be received on this interface, after the maximum number of un-acknowledged LSPs has been sent.</t>
+			<t>Type: 2</t>
+			<t>Length: 4 octets</t>
+			<t>Value: minimum interval, in micro-seconds, between two consecutive LSPs sent after the burst window has been used</t>
+			<t>The LSP Transmission Interval is an advertisement of the receiver's steady-state LSP reception rate.</t>
+		</section>
+		<section anchor="LPP" title="LSPs Per PSNP sub-TLV">
+			<t>The sub-TLV LSP Per PSNP (LPP) advertises the number of received LSPs that triggers the immediate sending of a PSNP to acknowledge them.</t>
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:374]: Change to: "The LSP per PSNP (LPP) sub-TLV ..."

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 4]
```diff
@@ -346,113 +356,138 @@
...
+				<t>Length: Indicates the length in octets (1-8) of the Value field. The length SHOULD be the minimum required to send all bits that are set.</t>
+				<t>Value: List of flags.</t>
+
+				<artwork align="left" pn="Flags">
+          0 1 2 3 4 5 6 7 ...
+         +-+-+-+-+-+-+-+-+...
+         |O|              ...
+         +-+-+-+-+-+-+-+-+...
+				</artwork>
+				<t>When the O flag is set, the LSP will be acknowledged in the order they are received: a PSNP acknowledging N LSPs is acknowledging the N oldest LSPs received. The order inside the PSNP is meaningless. If the sender keeps track of the order to LSP sent, this indication allows a fast detection of loss of LSP. This MUST NOT be used to trigger faster retransmission of LSP. This MAY be used to trigger a congestion signal.</t>
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:392]: s/to LSP sent/of LSPs sent/
s/this indication/the ordering/
s/a fast detection of loss of LSP/the fast detection of the loss of an LSP/
s/retransmmission of LSP/retransmission of an LSP/
s/to trigger a congestion signal/as an indication of a congestion event/

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 5]
```diff
@@ -346,113 +356,138 @@
...
+				<t>A given LSP transmitter will receive multiple flooding parameter advertisements from different receivers that may carry different flooding parameter values. A given transmitter SHOULD use the most convervative value on a per parameter basis. For example, if the transmitter receives multiple LSP Burst Window values, it should use the smallest value.</t>
+				<t>In order for the LSP Burst Window to be a useful parameter, an LSP transmitter needs to be able to keep track of the number of un-acknowledged LSPs it has sent to a given LSP receiver. On a LAN there is no explicit acknowledgment of the receipt of LSPs between a given LSP transmitter and a given LSP receiver. However, an LSP transmitter on a LAN can infer whether any LSP receiver on the LAN has requested retransmission of LSPs from the DIS by monitoring PSNPs generated on the LAN. If no PSNPs have been generated on the LAN for a suitable period of time, then an LSP transmitter can safely set the number of un-acknowledged LSPs to zero. Since this suitable period of time is much higher than the fast acknowledgment of LSPs defined in <xref target="LSPACKRate"/>
+,
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:400]: Moving the comma to start on a new line introduces an incorrect space.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 6]
```diff
@@ -466,54 +501,55 @@
...
+						<t>If either the LSP transmitter or receiver does not adhere to these parameters, for example because of transient conditions, this causes no fatal condition to the operation of IS-IS. In the worst case, an LSP is lost at the receiver and this situation is already remedied by mechanisms in <xref target="ISO10589"/>
+. After a few seconds, neighbors will exchange PSNPs (for point to point interfaces) or CSNPs (for broadcast interfaces) and recover from the lost LSPs. This worst case should be avoided as those additional seconds impact convergence time as the LSDB is not fully synchronized. Hence it is better to err on the conservative side and to under-run the receiver rather than over-run it.</t>
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:523]: Again, moving the period to a new line introduces an incorrect space.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 7]
```diff
@@ -547,162 +584,173 @@ if rtt_var == NA then rtt_var = rtt_est / 2
...
+						<t>If there is no transmitted LSP for a fixed period of time, e.g. 2 seconds, the sender switches back to the slow start phase.</t>
+					</section>
+					<section anchor="cc_remarks" title="Remarks">
+						<t>This algorithm's performance is dependent on the LPP value. Indeed, the smaller LPP is, the more information is available for the congestion control algorithm to perform well. However, it also increases the resources spent on sending PSNPs, so a tradeoff must be made. This document recommends to use an LPP of 15 or less.</t>
+
+						<t>Note that this congestion control algorithm benefits from the extensions proposed in this document. The advertisement of a receive window from the receiver (<xref target="FlowControl"/>
+) avoids the use of an arbitrary maximum value by the sender. The faster acknowledgment of LSPs (<xref target="LSPACKRate"/>
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:598]: Another bad line break resulting in bad formatting.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 8]
```diff
@@ -547,162 +584,173 @@ if rtt_var == NA then rtt_var = rtt_est / 2
...
+) allows for a faster control loop and hence a faster increase of the congestion window in the absence of congestion.
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:599]: Another bad line break resulting in bad formatting.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 9]
```diff
@@ -547,162 +584,173 @@ if rtt_var == NA then rtt_var = rtt_est / 2
...
+				<t>The values may also be absolute value reflecting relevant average hardware resources that are been monitored, typically the amount of buffer space used by incoming LSPs. In this case, care must be taken when choosing the parameters influencing the values in order to avoid undesirable or instable feedback loops. It would be undesirable to use a formula that depends, for example, on an active measurement of the instantaneous CPU load to modify the values advertised in the Flooding Parameters TLV. This could introduce feedback into the IGP flooding process that could produce unexpected behavior.</t>
+			</section>
 
-<section anchor="OPS_Considerations" title="Operation considerations">
-	<t>As discussed in  <xref target="TLVoperationLAN"/>
+			<section anchor="OPS_Considerations" title="Operation considerations">
+				<t>As discussed in  <xref target="TLVoperationLAN"/>
 , the solution is more effective on point to point adjacencies. Hence a broadcast interface (e.g. Ethernet) only shared by two IS-IS neighbhors should be configured as point to point in order to have a more effective flooding.</t>
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:613]: Another bad line break resulting in bad formatting.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 10]
```diff
@@ -547,162 +584,173 @@ if rtt_var == NA then rtt_var = rtt_est / 2
...
+			<t>The flow control algorithm MUST NOT assume the receive capabilities
         of a neighbor are static, i.e., it MUST handle transient conditions
         which result in a slower or faster receive rate on the part of a
         neighbor.</t>
 
-        <t>The flow control algorithm needs to consider the expected delay
-        time in receiving an acknowledgment. See <xref target="LSPACKRate"/>.
+			<t>The flow control algorithm needs to consider the expected delay
+        time in receiving an acknowledgment. See <xref target="LSPACKRate"/>
+.
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:654]: Another bad line break resulting in bad formatting.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 11]
```diff
@@ -547,162 +584,173 @@ if rtt_var == NA then rtt_var = rtt_est / 2
...
+			<c>4</c>
+			<c>Flags</c>
+			<c>5-255</c>
+			<c>Unassigned</c>
+		</texttable>
+	</section>
+	<section anchor="Security" title="Security Considerations" toc="default">
+		<t>
+    Security concerns for IS-IS are addressed in <xref target="ISO10589"/>
+,
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:695]: Another bad line break resulting in bad formatting.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 12]
```diff
@@ -547,162 +584,173 @@ if rtt_var == NA then rtt_var = rtt_est / 2
...
+		<xref target="RFC5304"/>
+, and <xref target="RFC5310"/>
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:697]: Another bad line break resulting in bad formatting.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 13]
```diff
@@ -547,162 +584,173 @@ if rtt_var == NA then rtt_var = rtt_est / 2
...
+.  These documents
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:698]: Another bad line break resulting in bad formatting.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml [hunk 14]
```diff
@@ -547,162 +584,173 @@ if rtt_var == NA then rtt_var = rtt_est / 2
...
     PDUs, including SNPs and IIHs. These authentication mechanisms are not
     altered by this document. 
-  </t>
-  <t>
+	</t>
+	<t>
     With the cryptographic mechanisms described in <xref target="RFC5304"/>
-    and <xref target="RFC5310"/>, an attacker wanting to advertise an incorrect
+    and <xref target="RFC5310"/>
+, an attacker wanting to advertise an incorrect
```
Comment (tony1athome) [inline draft-decraeneginsberg-lsr-isis-fast-flooding.xml:706]: Another bad line break resulting in bad formatting.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml (hunk 2 above)
Comment (Bruno007) [reply draft-decraeneginsberg-lsr-isis-fast-flooding.xml:214]: OK. Thank you

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml (hunk 10 above)
Comment (Bruno007) [reply draft-decraeneginsberg-lsr-isis-fast-flooding.xml:654]: Not sure where this is coming from but corrected.

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml (hunk 11 above)
Comment (Bruno007) [reply draft-decraeneginsberg-lsr-isis-fast-flooding.xml:695]: ack

File: draft-decraeneginsberg-lsr-isis-fast-flooding.xml (hunk 14 above)
Comment (Bruno007) [reply draft-decraeneginsberg-lsr-isis-fast-flooding.xml:706]: ack

**Pull request merged**